src/Microsoft.DotNet.Build.Tasks/DownloadFileFromUrl.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class DownloadFileFromUrl : BuildTask
+    {
+        /// <summary>
+        /// The url to download the file from.
+        /// </summary>
+        [Required]
+        public string DownloadSource { get; set; }
+        
+        /// <summary>
+        /// The directory where the file will be downloaded to.
+        /// </summary>
+        [Required]
+        public string DestinationFile { get; set; }
+
+        /// <summary>
+        /// The default is to fail on error, set this to true if you want to just warn.
+        /// </summary>
+        public bool TreatErrorsAsWarnings { get; set; }
+
+        public override bool Execute()
+        {
+            bool success = false;
+            try
+            {
+                Log.LogMessage(MessageImportance.High, $"Downloading {DownloadSource} -> {DestinationFile}");
```

Comment:
NIT: do we really want this to be `MessageImportance.High`? I would expect this type of messages to not be present on minimal logging.

Reply:
I made it like this because downloading a file can take ~5 or ~10 seconds, so that when we are building we know we are downloading something, but I can remove it and leave it to the targets to log that.

Reply:
that is true for this specific use because it's downloading a nuget package, but this task can/will be reused to download other things in the future, and I'm not sure if we will always want to have the verbosity in all cases. That's why I would suggest to keep logging a message, just not make it Importance.High

Reply:
Sounds reasonable. We could also pass an optional parameter to enable High verbosity and log with Importance.High if that is enabled.

Reply:
I wouldn't do that. Just make the logging normal verbosity and if folks want that information they can always enable that level of logging.

Reply:
Ok, sounds good. Will do that then.